searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp

```diff
@@ -720,13 +700,19 @@ class CreateBlueprintVisitor : public CreateBlueprintVisitorHelper
             NodeAsKey key(n, _scratchPad);
             setResult(std::make_unique<DirectAttributeBlueprint>(_field, _attr.getName(), _attr, *_dwa, key));
         } else {
+            attribute::SearchContextParams scParams;
+            scParams.useBitVector(_field.isFilter())
+                    .metaStoreReadGuard(getRequestContext().getMetaStoreReadGuard());
```

Comment:
To avoid forgetting to propagate the meta store read guard, consider adding a helper function that creates the baseline SearchContextParams (with meta store read guard set), and use this function all places where _setResult()_ is called.

Reply:
Done